### ISSUE TITLE
pytorch generate.py invoked on tensorflow generated pkl generates different image for same seed. Expected?

### ISSUE THREAD

**Issue by MoemaMike** [NVlabs/stylegan2-ada-pytorch#83]
I trained a network very recently using stylegan2-ada. I noticed that many of my images generated by generate.py in the tensorflow version have consistent annoying imperfections in the top row(s) of pixels, they appear distractingly lighter than they should be compared to the pixels in the rest of the generated image. This creates a distracting lighter line at the top of the image. 

Not knowing whether this was a problem with the network or with the generator, and knowing that the tensorflow code is deprecated,  I thought that it might be the generator and therefor I might be able to get a of the same image(s) without these visual artifacts by using the pytorch version of generator for the same seeds. But what i found is that when i run pytorch generate against the tensorflow created pkl file for the same seed that i had used in tensorflow, I do not get the exact same image as i was expecting. I get an image that is clearly visually related to the tensflorflow image for that seed, to the extent that the two images are composed of visually similar colors and such, but the figurative elements are jumbled around. See attached 2 images, 1 generated by tensorflow and 1by pytorch as indicated by file names, for the same seeds.

![seed0050-tensorflow-generate](https://user-images.githubusercontent.com/81262214/113362251-550db700-9324-11eb-9f90-0410b612d600.png)

![seed0050-pytorch-generate](https://user-images.githubusercontent.com/81262214/113362240-50490300-9324-11eb-9311-4c2f40063605.png)

So is this expected? Is there no way for me to generate the exact same image? Btw the example attached is not one of the images that hd the row(s) of light pixels at the top. that is a separate issue.

Assuming this is expected behavior help me to understand the implications.  I have a single pkl file which is portable across the stylegan2-ada and stylegan2-ada-pytorch impls, but for the same seeds, each generator will generate different sets of images. does that mean that the "latent spaces" are different in the 2 impls or that the same seed will point to a different "location" in the same latent space? Another way of asking .. is the exact same image generatable by both impls but for unknowably different seeds?

**Comment by nurpax** [comment]
I'd guess that the difference is caused by the noise inputs not matching between TensorFlow and PyTorch versions.

Search for `noise_mode` in style_mixing.py and `noise_bufs` in projector.py for some ideas on how its used.  You could probably match TF and PyTorch if you'd generate this noise yourself with numpy using a fixed seed and then feeding it in.  The code for doing that is somewhat tricky but you can get the general idea by reading projector.py.

**Comment by MoemaMike** [comment]
Thanks for the reply. That sounds logical and worth looking into for this specific issue.

Aside, after generating more images with pytorch generator, i have found that many of the images have the same problem with distractingly light pixels at the upper border, which was my original motivation for attempting to generate the same image with both generators. I will file a separate issue for that.
